File: 155-MinStack/155-MinStack.py
from collections import deque
class MinStack:

    def __init__(self):
        self.stack = []
        self.min = []

    def push(self, val: int) -> None:
        if len(self.min) == 0 or self.min and val < self.min[-1]:
            self.min.append(val)
        else:
            self.min.append(self.min[-1])   
    
        self.stack.append(val)


    def pop(self) -> None:
        self.stack.pop()
        self.min.pop()

        

    def top(self) -> int:
        return self.stack[-1] if self.stack else 0
        

    def getMin(self) -> int:
        return self.min[-1] if self.min else 0

        


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()

File: 155-MinStack/155-MinStack.js

var MinStack = function() {
    this.stack = [];
    this.min = [];
};

/** 
 * @param {number} val
 * @return {void}
 */
MinStack.prototype.push = function(val) {
    if(this.min.length == 0 || this.min.length > 0 && val < this.min[this.min.length - 1] ){
        this.min.push(val);
    } else {
        this.min.push(this.min[this.min.length - 1]); 
    }
    this.stack.push(val);
};

/**
 * @return {void}
 */
MinStack.prototype.pop = function() {
    this.min.pop();
    this.stack.pop();
};

/**
 * @return {number}
 */
MinStack.prototype.top = function() {
    return this.stack[this.stack.length - 1]
};

/**
 * @return {number}
 */
MinStack.prototype.getMin = function() {
    return this.min[this.min.length - 1]
};

/** 
 * Your MinStack object will be instantiated and called as such:
 * var obj = new MinStack()
 * obj.push(val)
 * obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.getMin()
 */
